### PR TITLE
[bugfix] costing removed as params for mapbox_osrm

### PR DESCRIPTION
**Pull request by RemaniTinpo**

### CODE REVIEW

File: tests/test_mapbox_osrm.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ def test_full_isochrones(self):
 
         self.assertEqual(1, len(responses.calls))
         self.assertURLEqual(
-            'https://api.mapbox.com/isochrone/v1/mapbox/driving/8.34234,48.23424?costing=mapbox/driving&access_token=sample_key&'
+            'https://api.mapbox.com/isochrone/v1/mapbox/driving/8.34234,48.23424?driving&access_token=sample_key&'
```
Comment (nilsnolde) [inline tests/test_mapbox_osrm.py:131]: it passes the test but it's still not really correct:

```suggestion
            'https://api.mapbox.com/isochrone/v1/mapbox/driving/8.34234,48.23424?access_token=sample_key&'
```

**Review by nilsnolde — COMMENTED** [review]
Thanks for the PR! Just a small oversight.

Actually we should also unify the interface: `Isochrones` expects different profile parameters (`mapbox/xxx`) while the other methods expect it without `mapbox/`, so we should adapt `isochrones` to use the same schema (ideally backwards-compatible, so that older code still works with `mapbox/xxx` after the update). If you don't have the time for it, I'll do it after merging this PR.

**Comment by RemaniTinpo** [comment]
Hi @nilsnolde,
Does your comment is linked to the second point of my issue ?

**Comment by nilsnolde** [comment]
Jep, right. We can leave the readme as-is and change the code to not require `mapbox/xxx`, just like the other MapboxOSRM functions. Was also an oversight apparently. Just gotta be a little careful to do it backwards-compatible, e.g. `profile.replace('mapbox/', '')` in the isochrones function, so old code will continue to work.

**Comment by RemaniTinpo** [comment]
Sorry, it was a busy week. Let me know if it's ok like this, or if you prefer something more.

File: routingpy/routers/mapbox_osrm.py [hunk 2]
```diff
@@ -501,6 +504,8 @@ def matrix(
         if fallback_speed:
             params['fallback_speed'] = str(fallback_speed)
 
+        profile = profile.replace('mapbox/','')
```
Comment (nilsnolde) [inline routingpy/routers/mapbox_osrm.py:507]: I believe this is superfluous, it's been without the `mapbox/` modifier before as well.

**Review by nilsnolde — CHANGES_REQUESTED** [review]
Just one minor change request, other than that it looks good, thanks @RemaniTinpo . And if you could add a short entry in the `FIXED`  section of the CHANGELOG, that'd be great.

**Comment by nilsnolde** [comment]
We'd like to release pretty soon and this fix would be nice to have. @RemaniTinpo if you wanna finish this would be cool, if no time, I'll add the missing pieces, no worries.

File: routingpy/routers/mapbox_osrm.py [hunk 3]
```diff
@@ -288,6 +288,8 @@ def directions(  # noqa: C901
 
         get_params = {"access_token": self.api_key} if self.api_key else {}
 
+        profile = profile.replace('mapbox/','')
```
Comment (nilsnolde) [inline routingpy/routers/mapbox_osrm.py:291]: didn't see that before, same comment as for matrix

**Review by nilsnolde — CHANGES_REQUESTED** [review]
great thx @RemaniTinpo 

in the meantime we changed quite a few things in master, e.g. now we have black as style checker instead of yapf. if you want to update this (CI needs to pass), I leave the instructions here, should be quick:

```
git merge upstream/master  # if you that's what our repo's remote is called for you
# likely resolve some conflicts
source .venv/bin/activate
poetry install  # to install the new deps and update others
pre-commit install  # install the new pre-commit hooks
```

if you then commit the next time black will auto-format your code, which you'll have to `git add` again and finally commit & push.

**Comment by RemaniTinpo** [comment]
Thanks, I didn't catched the move to black. It's very nice btw you changed for black.
And sorry for the delay, awfull week, I will stay tuned today !

File: routingpy/routers/mapbox_osrm.py [hunk 4]
```diff
@@ -412,9 +411,13 @@ def isochrones(
         if generalize:
             params["generalize"] = generalize
 
+        profile = profile.replace("mapbox/", "")
+
         return self._parse_isochrone_json(
-            self.client._request(
-                "/isochrone/v1/" + profile + "/" + locations_string, get_params=params, dry_run=dry_run
```
Comment (nilsnolde) [inline routingpy/routers/mapbox_osrm.py:417]: we also changed the way we request, in order to be able to switch the client class for other purposes (we need that for other projects). you'll only have to change `self._request` to `self.client._request`.

**Review by nilsnolde — CHANGES_REQUESTED** [review]
Looks good! Last little thing to get past the tests and this is done @RemaniTinpo 

Thanks for your patience here.

File: routingpy/routers/mapbox_osrm.py (hunk 4 above)
Comment (RemaniTinpo) [reply routingpy/routers/mapbox_osrm.py:417]: I was writing a suggest change, but you have to been to fast :)

**Comment by nilsnolde** [comment]
jep that's it:)

**Pull request merged**